lib/usd/ui/layerEditor/layerTreeItem.cpp

```diff
@@ -267,13 +267,21 @@ bool LayerTreeItem::isIncoming() const { return _isIncomingLayer; }
 
 bool LayerTreeItem::needsSaving() const
 {
-    if (_layer) {
-        if (!isSessionLayer() && !isReadOnly()
```

Comment:
The removal of the call to isReady() is the only real change, all the other is just avoiding the complex nesting of conditionals and multi-line conditions, allowing to document what each condition does.

Reply:
So if I understand it correctly, the needsSaving() function will return true even if `permissionToSave` is false, and that will be checked later when actually writing the layer

Reply:
NO, isReadOnly checks permission to edit, not save. We currently do not have check for permission to save, but then again, USD itself will refuse to save, so we don't need to check it to prevent it. When we do the future work to have a UI for permission to save, we might had new verifications.